# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../iproute2/0004-include-sys-time.h.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

include sys/time.h for getting struct timeval

------------------------------------------------------------------------------
In file included from lnstat.c:39:0:
lnstat.h:28:17: error: field 'last_read' has incomplete type
lnstat.h:29:17: error: field 'interval' has incomplete type
------------------------------------------------------------------------------

misc/lnstat.h:
------------------------------------------------------------------------------
struct lnstat_file {
        struct lnstat_file *next;
        char path[PATH_MAX+1];
        char basename[NAME_MAX+1];
        struct timeval last_read;               /* last time of read */
        struct timeval interval;                /* interval */
        int compat;                             /* 1 == backwards compat mode */
        FILE *fp;
        unsigned int num_fields;                /* number of fields */
        struct lnstat_field fields[LNSTAT_MAX_FIELDS_PER_LINE];
};
------------------------------------------------------------------------------

--- a/misc/lnstat.c	2013-08-31 21:04:17.547887885 +0200
+++ b/misc/lnstat.c	2013-08-31 21:04:39.352378601 +0200
@@ -35,6 +35,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <getopt.h>
+#include <sys/time.h>
 
 #include "lnstat.h"
 
